nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py

```diff
@@ -93,125 +201,241 @@ def get_predicted_dialog_ret_sys_act(dialog, all_predictions, schemas, eval_debu
+                        )
+
+                    if predictions["noncat_slot_status_GT"][slot_idx] != STATUS_OFF:
+                        noncat_slot_value_num += 1
+                        if ext_value is not None and ext_value in true_state['slot_values'][slot]:
+                            noncat_slot_value_acc += 1
 
-                        if ch_start_idx > 0 and ch_end_idx > 0:
-                            # Add span from the user utterance.
-                            slot_values[slot] = user_utterance[ch_start_idx - 1 : ch_end_idx]
-                        else:
-                            if slot in sys_prev_slots[frame["service"]]:
-                                # debugging info
-                                sys_rets[slot] = sys_prev_slots[frame["service"]][slot]
-                                slot_values[slot] = sys_prev_slots[frame["service"]][slot]
+                    if predictions["noncat_slot_status_GT"][slot_idx] == predictions["noncat_slot_status"][slot_idx]:
+                        noncat_slot_status_acc += 1
+                    # debugging info processing ended
+
+                carry_over_slots(
+                    frame,
+                    all_slot_values,
+                    schemas.slots_relation_list,
+                    frame_service_prev,
+                    slot_values,
+                    sys_slots_agg,
+                    sys_slots_last,
+                )
 
                 if eval_debug and frame["service"] in in_domain_services:
-                    logging.debug("-----------------------------------New Frame------------------------------")
-                    logging.debug(f'SYS : {system_utterance}')
-                    logging.debug(f'USER: {user_utterance}')
-
-                    logging.debug("\n")
-                    logging.debug(f"PRED CAT: {categorical_slots_dict}")
-                    logging.debug(f"PRED NON-CAT: {non_categorical_slots_dict}")
-
-                    logging.debug("\n")
-                    logging.debug(f"SLOTS - LABEL: {true_slots}")
-                    logging.debug(f"STATE - LABEL: {true_state['slot_values']}")
-                    logging.debug(f"STATE - PRED : {slot_values}")
-
-                    logging.debug("\n")
-                    logging.debug(f"SYS PREV SLOT: {sys_prev_slots}")
-                    logging.debug(f"SYS RETS: {sys_rets}")
-                    cat_slot_status_acc = (
-                        "NAN" if cat_slot_status_num == 0 else cat_slot_status_acc / cat_slot_status_num
-                    )
-                    logging.debug(f"CAT STATUS ACC: {cat_slot_status_acc}")
-                    noncat_slot_status_acc = (
-                        "NAN" if noncat_slot_status_num == 0 else noncat_slot_status_acc / noncat_slot_status_num
-                    )
-                    logging.debug(f"NONCAT STATUS ACC: {noncat_slot_status_acc}")
+                    equal_state = True
```

Comment:
Could you add some comments on what is happening here?

Reply:
Done.